apps/emqx_management/src/emqx_mgmt_api_publish.erl

```diff
@@ -42,23 +46,46 @@ schema("/publish") ->
     #{
         'operationId' => publish,
         post => #{
-            description => <<"Publish Message">>,
+            description => <<
+                "Publish one message<br/>"
+                "Possible HTTP status response codes are:<br/>"
+                "200: The message is delivered to at least one subscriber<br/>"
+                "202: No matched subscriber<br/>"
+                "400: Message is invalid. for example bad topic name or QoS out of range<br/>"
+                "503: Failed to deliver the message to subscriber(s)<br/>"
+            >>,
             tags => [<<"Publish">>],
             'requestBody' => hoconsc:mk(hoconsc:ref(?MODULE, publish_message)),
             responses => #{
-                200 => hoconsc:mk(hoconsc:ref(?MODULE, publish_message_info))
+                200 => hoconsc:mk(hoconsc:ref(?MODULE, publish_ok)),
+                202 => hoconsc:mk(hoconsc:ref(?MODULE, publish_error)),
+                400 => hoconsc:mk(hoconsc:array(hoconsc:ref(?MODULE, publish_error)), #{}),
+                ?DISPATCH_ERROR => hoconsc:mk(hoconsc:ref(?MODULE, publish_error))
             }
         }
     };
 schema("/publish/bulk") ->
     #{
         'operationId' => publish_batch,
         post => #{
-            description => <<"Publish Messages">>,
+            description => <<
+                "Publish a batch of messages<br/>"
+                "Possible HTTP response status code are:<br/>"
+                "200: All message are delivered to at least one subscriber<br/>"
+                "202: At least one message matched no subscriber<br/>"
+                "400: If there is at least one invalid message "
+                "for example bad topic name or QoS out of range<br/>"
```

Comment:
"At least one of the messages is invalid. 
For example bad topic name or QoS is out of range"